src/ModelIO/MDLAsset.cs

```diff
@@ -1,12 +1,23 @@
 #if XAMCORE_2_0 || !MONOMAC
 using System;
+using XamCore.ObjCRuntime;
 namespace XamCore.ModelIO {
 	public partial class MDLAsset {
 		public MDLObject this [nuint index] {
 			get {
 				return GetObject (index);
 			}
 		}
+
+		[iOS (10,3), TV (10,2), Mac (10,12,4)]
+		public IMDLComponent this [Protocol key] {
```

Comment:
Mmh here's the guideline: https://msdn.microsoft.com/en-us/library/ms229006(v=vs.110).aspx

> **AVOID** indexers with parameter types other than System.Int32, System.Int64, System.String, System.Object, or an enum.

So `Protocol` is definitely not recommended.

> **DO NOT** provide more than one family of overloaded indexers in one type.

I guess that validates your point @spouliot. As I understand it it's fine to "overload" an indexer (as in have an indexer that's using an other indexer in the same type) but it's wrong to create a 2nd and different family of indexers.